compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -783,7 +783,7 @@ object desugar {
         DefDef(
           className.toTermName, joinParams(constrTparams, defParamss),
           classTypeRef, creatorExpr)
-          .withMods(companionMods | mods.flags.toTermFlags & GivenOrImplicit | Synthetic | Final)
+          .withMods(companionMods | mods.flags.toTermFlags & GivenOrImplicit | Final)
```

Comment:
could an alternative be that in `Namer` we also look up specifically `isAllOf(Synthetic|Given, butNot = Private)` when looking up wildcard exports? then there is no need for TASTy changes.

Reply:
se PR #12999

Reply:
That does not solve it since it still keeps the Synthetic flag on the given or implicit def. But I think I have it figured out now.